File: PAT_b_1032.py
#N = int(input())
#result = {}
#for i in range(N):
#	num,score = tuple(map(int,input().split(" ")))
#	if num in result.keys():
#		result[num] += score
#	else:
#		result[num] = score
#M = max(result.items(),key = lambda x:x[1])
#print(M[0],M[1])

N = int(input())
result = [0]*N
for i in range(N):
	num,score = tuple(map(int,input().split(" ")))
	result[num-1] += score
Max = 0
indx = 0
for i in range(N):
	if result[i] >= Max:
		Max = result[i]
		indx = i
print(indx+1,Max)

File: PAT_b_1032.cc
#include <iostream>

int main(){
	int N=0;
	std::cin >> N;
	int num,score;
//	int result[N]=0;
	int *result = new int[N+2] ();
	for(int i=0;i<N;++i){
		std::cin >> num >> score;
		result[num] += score;
	}
	int max = -1,indx = 0;
	for(int i=0;i<N+2;++i){
		if( result[i]>=max){
			max = result[i];
			indx = i;
		}
	}
	std::cout<<indx<<" "<<max<<std::endl;
}

